### PR TITLE
Don't re-run workflows on forks.

### PR DESCRIPTION
**Pull request by RandomDSdevel**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

~~After making all changes to a cask, verify:~~  

- [ ] ~~`brew cask audit --download {{cask_file}}` is error-free.~~
- [ ] ~~`brew cask style --fix {{cask_file}}` reports no offenses.~~
- [ ] ~~There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.~~  
- [ ] ~~The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).~~

(N/A; not modifying any casks.)  

~~Additionally, **if adding a new cask**:~~  

- [ ] ~~Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).~~  
- [ ] ~~`brew cask audit --new-cask {{cask_file}}` worked successfully.~~  
- [ ] ~~`brew cask install {{cask_file}}` worked successfully.~~  
- [ ] ~~`brew cask uninstall {{cask_file}}` worked successfully.~~  
- [ ] ~~Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).~~  
- [ ] ~~Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).~~  

(N/A; not adding any new casks.)  

Please verify:  

- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same change.  

---

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;After:  

 - Coming back to working with Homebrew and Homebrew Cask after an extended hiatus, 
 - Pulling any new changes down from GitHub, and
 - Pushing them to my personal forks, 

I noticed CI jobs were getting triggered on my personal Homebrew Cask fork by the 'rerun-workflow' Actions workflow.  This PR is a naïve attempt at fixing that; feel free to request or add changes or supersede as needed/desired.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ env:
 
 jobs:
   generate-matrix:
+    if: startsWith(github.repository, "Homebrew/cask")
```
Comment (vitorgalvao) [inline .github/workflows/ci.yml:12]: There isn’t an `Homebrew/cask` repository, it’s `Homebrew/homebrew-cask`.

**Review by vitorgalvao — DISMISSED** [review]

**Comment by vitorgalvao** [comment]
You don’t need to strikethrough the items, just don’t fill the checkbox.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (reitermarkus) [reply .github/workflows/ci.yml:12]: Make this just `Homebrew/`.

**Comment by reitermarkus** [comment]
I don't think it's needed for the CI job, since this will run only on pull requests anyways.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (RandomDSdevel) [reply .github/workflows/ci.yml:12]: @vitorgalvao:  

> There isn’t an `Homebrew/cask` repository, it’s `Homebrew/homebrew-cask`.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(_Facepalms._)  That's what I get for using `brew --repository` (and other `brew` tap-related commands,) where that shortcut applies, too much.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;I've addressed all comments and rebased against `master`.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whoops, _now_ I've rebased.

**Review by reitermarkus — APPROVED** [review]

**Comment by RandomDSdevel** [comment]
@reitermarkus, @vitorgalvao:  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;After keeping an eye on things for a bit, I'm seeing on my fork that my changes didn't _quite_ stop workflows from re-running there.  I'm going to take another look at the notifications I get from GitHub Actions both before and after this PR, report my findings back, and see if I can't figure out what's going on so I can fix it with another PR.  (I'll just disable Actions on my fork in its settings if I have to, but I'd rather fix this for everyone, not just me.  )

**Comment by reitermarkus** [comment]
I think `"Homebrew/"` needs to be single quoted.

**Comment by RandomDSdevel** [comment]
@reitermarkus:  

> I think "Homebrew/" needs to be single-quoted.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Makes sense; I thought a trip to GitHub Actions's syntax documentation might be in order anyway if nothing obvious popped out at me…

**Pull request merged**